torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -728,18 +742,20 @@ def _init_reshard_after_forward(self):
 
     def _get_ignored_modules(
         self,
+        root_module: torch.nn.Module,
         _ignored_modules: Any,
     ) -> Set[torch.nn.Module]:
         """
-        Checks that ``_ignored_modules`` is an iterable of ``torch.nn.Module``
-        s without any :class:`FullyShardedDataParallel` instances and then
-        returns them and their children as a :class:`set`, excluding nested
-        :class:`FullyShardedDataParallel` instances.
+        Checks that ``_ignored_modules`` (1) is an iterable of
+        ``torch.nn.Module`` s without any :class:`FullyShardedDataParallel`
+        instances and does not contain the top-level ``module`` itself, and
+        then returns them and their children as a :class:`set`, excluding
+        nested :class:`FullyShardedDataParallel` instances.
 
-        We include the child modules to better match user intuition since
-        ignoring a module should ignore its child modules as well, and we
-        exclude :class:`FullyShardedDataParallel` instances since ``self`` may
-        be the intended root instance that manages them.
+        We include the child modules of modules in ``_ignored_modules`` to be
+        more intuitive since ignoring a module should ignore its child modules
+        as well, and we exclude :class:`FullyShardedDataParallel` instances
+        since ``self`` may be the intended root instance that manages them.
```

Comment:
I feel like if a `module` contains a nested FSDP instance and a user wants to ignore `module`, something has gone wrong because portions of it will still be sharded? Or do you anticipate this being a use case?

Reply:
Good point. I definitely have less experience with how users may want to use the `ignore_modules` argument, so I trust your intuition here. Though, I think this is something we can revisit since the only change here would be to error instead of ignore.